### PR TITLE
connection secrets: log when a connection secret cannot be fetched

### PR DESCRIPTION
**Pull request by suskin**
### Description of your changes

If a resource asks for a connection secret which doesn't exist (for
example, if the resource is misconfigured), the code will silently
ignore the secret. This change adds a log line so that it is easier to
tell when this happens.

This change is related to #331 and #597 .

I opted to go with debug logging for now, but an argument could be made that this should be a loud error in the logs instead.

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [x] Updated any relevant [documentation], [examples], or [release notes].
- [x] Updated the RBAC permissions in [`clusterrole.yaml`] to include any new types.

[documentation]: https://github.com/crossplaneio/crossplane/tree/master/docs
[examples]: https://github.com/crossplaneio/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplaneio/crossplane/tree/master/PendingReleaseNotes.md
[`clusterrole.yaml`]: https://github.com/crossplaneio/crossplane/blob/master/cluster/charts/crossplane/templates/clusterrole.yaml

### CODE REVIEW

**Comment by negz** [comment]
> If a resource asks for a connection secret which doesn't exist (for example, if the resource is misconfigured), the code will silently ignore the secret.

Do we really ignore it, or do we just not log it? The below line, right around where you added the log, appears to be updating the conditioned status of the resource under reconciliation to reflect the same error we're now logging.

```go
ar.Status.SetConditions(corev1alpha1.ReconcileError(err))
```

I'm a bit of a broken record at this point, but in many organisations the person trying to use the broken `KubernetesApplication` will likely not be the same person running the Crossplane infrastructure. In such scenarios I believe it's a better user experience to `kubectl describe thebrokenthing` to triage it than to raise a ticket to have your Crossplane administrator go look at the logs.

**Comment by suskin** [comment]
> > If a resource asks for a connection secret which doesn't exist (for example, if the resource is misconfigured), the code will silently ignore the secret.
> 
> Do we really ignore it, or do we just not log it? The below line, right around where you added the log, appears to be updating the conditioned status of the resource under reconciliation to reflect the same error we're now logging.
> 
> ```go
> ar.Status.SetConditions(corev1alpha1.ReconcileError(err))
> ```
> 
> I'm a bit of a broken record at this point, but in many organisations the person trying to use the broken `KubernetesApplication` will likely not be the same person running the Crossplane infrastructure. In such scenarios I believe it's a better user experience to `kubectl describe thebrokenthing` to triage it than to raise a ticket to have your Crossplane administrator go look at the logs.

Sorry about not being clear. Here's what the conditions look like for a wordpress deployment affected by this. I got this output by creating the resources in the `app.yaml` for the wordpress example using [the hash](https://github.com/crossplaneio/crossplane/commit/c9c71f82cfcb746491b5207000f6ba96f4487a4d) mentioned in #597, and then running `kubectl describe kubernetesapplicationresource -n complex wordpress-demo-deployment`.

```
  Conditions:
    Last Transition Time:  2019-07-26T16:46:14Z
    Reason:                Successfully reconciled managed resource
    Status:                True
    Type:                  Synced
  Remote:
    Conditions:
      Last Transition Time:  2019-07-26T16:18:26Z
      Last Update Time:      2019-07-26T16:18:26Z
      Message:               Deployment does not have minimum availability.
      Reason:                MinimumReplicasUnavailable
      Status:                False
      Type:                  Available
      Last Transition Time:  2019-07-26T16:28:27Z
      Last Update Time:      2019-07-26T16:28:27Z
      Message:               ReplicaSet "wordpress-8545774bcf" has timed out progressing.
      Reason:                ProgressDeadlineExceeded
      Status:                False
      Type:                  Progressing
```

The short version is that the reconcile error never shows up for the user, probably because the error is not fatal, and later in the reconciliation pass, the reconcile is recorded as being successful (as hypothesized in #597).

I am not saying that this change fixes the underlying issue, because the error would ideally be more obvious to the end user . . . #597 is probably the best place to discuss potential solutions for the problem. However, I am suggesting that adding this log line is better than what we currently have, because then the error shows up somewhere.

Did that answer the question @negz ?

**Comment by suskin** [comment]
@negz that makes sense. Yes, I agree that a hard fail would be more obvious for the user, and I would go with that approach unless a partial success is a good way to degrade gracefully in the face of a different type of failure. When I found #555, my initial impression was that #555 could also be a reason to `continue` instead of making the error fatal, but maybe it isn't a good reason.
